File: Fly_scan/app/src/main/java/com/example/fly_scan/MainActivity.kt
package com.example.fly_scan

import android.annotation.SuppressLint
import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.content.IntentFilter
import android.os.Bundle
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    companion object {
        const val m_Broadcastname: String = "com.barcode.sendBroadcast"
        lateinit var tv_code: TextView
    }

    private val receiver = MyCodeReceiver()
    //private lateinit var sm: ScanManager

    //@SuppressLint("WrongConstant")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        tv_code = this.findViewById(R.id.tv_showCode)

//        sm= getSystemService("olc_service_scan") as ScanManager
//        sm.setScanSwitchLeft(true)
//        sm.setScanSwitchRight(true)
//        sm.setBarcodeReceiveModel(2)


        tv_code.setOnClickListener {
            val intent = Intent()
            intent.action = "com.barcode.sendBroadcastScan"
            sendBroadcast(intent)
        }

    }

    override fun onResume() {
        super.onResume()
        registerBroadcast()
    }

    private fun registerBroadcast() {
        val intentFilter = IntentFilter()
        intentFilter.addAction(m_Broadcastname)
        registerReceiver(receiver, intentFilter)
    }

    class MyCodeReceiver : BroadcastReceiver() {
        override fun onReceive(context: Context, intent: Intent) {
            if (intent.action == m_Broadcastname) {
                val str = intent.getStringExtra("BARCODE")
                if ("" != str) {
                    tv_code.text = str
                }
            }
        }

        companion object {
            private const val TAG = "MycodeReceiver"
        }
    }

    override fun onPause() {
        super.onPause()
        unregisterReceiver(receiver)
    }

}

File: TestApp/MainActivity.cs
using System;
using Android.App;
using Android.Content;
using Android.OS;
using Android.Runtime;
using Android.Support.Design.Widget;
using Android.Support.V7.App;
using Android.Views;
using Android.Widget;

namespace TestApp
{
    [Activity(Label = "@string/app_name", Theme = "@style/AppTheme.NoActionBar", MainLauncher = true)]
    public class MainActivity : AppCompatActivity
    {
        public const string broadcastName = "com.barcode.sendBroadcast";
        /// <summary>
        /// 测试环境获取,正式环境不建议使用静态获取
        /// </summary>
        public static MainActivity Instance;
         
        public FlxScanCodeReceiver flxScanCodeReceiver = new FlxScanCodeReceiver();

        protected override void OnCreate(Bundle savedInstanceState)
        {
            base.OnCreate(savedInstanceState);
            Xamarin.Essentials.Platform.Init(this, savedInstanceState);
            SetContentView(Resource.Layout.activity_main);
            Instance = this;

            Android.Support.V7.Widget.Toolbar toolbar = FindViewById<Android.Support.V7.Widget.Toolbar>(Resource.Id.toolbar);
            SetSupportActionBar(toolbar);

            FloatingActionButton fab = FindViewById<FloatingActionButton>(Resource.Id.fab);
            fab.Click += FabOnClick;
            Com.Scan.Flx.FlxScan.Init(this);
        }

        protected override void OnResume()
        {
            base.OnResume();
            var intentFilter = new IntentFilter();
            intentFilter.AddAction(broadcastName);
            RegisterReceiver(flxScanCodeReceiver, intentFilter);
        }

        protected override void OnPause()
        {
            base.OnPause();
            UnregisterReceiver(flxScanCodeReceiver);
        }

        public override bool OnCreateOptionsMenu(IMenu menu)
        {
            MenuInflater.Inflate(Resource.Menu.menu_main, menu);
            return true;
        }

        public override bool OnOptionsItemSelected(IMenuItem item)
        {
            int id = item.ItemId;
            if (id == Resource.Id.action_settings)
            {
                return true;
            }

            return base.OnOptionsItemSelected(item);
        }

        private void FabOnClick(object sender, EventArgs eventArgs)
        {
            Intent intent = new Intent();
            intent.SetAction("com.barcode.sendBroadcastScan");
            SendBroadcast(intent);
        }
        public override void OnRequestPermissionsResult(int requestCode, string[] permissions, [GeneratedEnum] Android.Content.PM.Permission[] grantResults)
        {
            Xamarin.Essentials.Platform.OnRequestPermissionsResult(requestCode, permissions, grantResults);

            base.OnRequestPermissionsResult(requestCode, permissions, grantResults);
        }
    }

    public class FlxScanCodeReceiver : BroadcastReceiver
    {
        public override void OnReceive(Context context, Intent intent)
        {
            if (intent.Action == MainActivity.broadcastName)
            {
                var str = intent.GetStringExtra("BARCODE");
                if ("" != str)
                {
                    Toast.MakeText(MainActivity.Instance, str, ToastLength.Long).Show();
                }
            }
        }
    }


}

